executor/src/witgen/jit/interpreter.rs

```diff
@@ -203,63 +229,94 @@ impl<T: FieldElement> EffectsInterpreter<T> {
 
         let row_offset: i64 = data.row_offset.try_into().unwrap();
         let mut eval_stack = vec![];
-        for action in &self.actions {
-            match action {
-                InterpreterAction::AssignExpression(idx, e) => {
-                    let val = e.evaluate(&mut eval_stack, &vars[..]);
-                    vars[*idx] = val;
-                }
-                InterpreterAction::ReadCell(idx, c) => {
-                    vars[*idx] = data
-                        .data
-                        .get((row_offset + c.row_offset as i64).try_into().unwrap(), c.id)
-                        .0;
-                }
-                InterpreterAction::ReadFixedColumn(idx, c) => {
-                    let poly_id = PolyID {
-                        id: c.id,
-                        ptype: PolynomialType::Constant,
-                    };
-                    vars[*idx] = fixed_data.fixed_cols[&poly_id].values_max_size()
-                        [usize::try_from(row_offset + c.row_offset as i64).unwrap()];
-                }
-                InterpreterAction::ReadParam(idx, i) => {
-                    vars[*idx] = get_param(params, *i);
-                }
-                InterpreterAction::WriteCell(idx, c) => {
-                    data.data.set(
-                        (row_offset + c.row_offset as i64).try_into().unwrap(),
-                        c.id,
-                        vars[*idx],
-                    );
-                }
-                InterpreterAction::WriteParam(idx, i) => {
-                    set_param(params, *i, vars[*idx]);
-                }
-                InterpreterAction::MachineCall(id, arguments) => {
-                    // we know it's safe to escape the references here, but the compiler doesn't, so we use unsafe
-                    let mut args = arguments
-                        .iter()
-                        .map(|a| match a {
-                            MachineCallArgumentIdx::Unknown(idx) => {
-                                let var = &mut vars[*idx] as *mut T;
-                                LookupCell::Output(unsafe { var.as_mut().unwrap() })
+        let mut block_stack = vec![self.actions.iter()];
```

Comment:
handling of the existing actions didn't change, its only code for handling the `Branch` action that is new